### PR TITLE
Add configurable conversion columns to sessions table

### PR DESCRIPTION
**Pull request by rlh1994**
## Description & motivation
This work adds configurable conversion columns to the sessions table in such a way that the user can specify the name and condition of the conversion, then optionally provide the the value of each conversion and if they would like each event id listed. These columns get added automatically as new values are added to the variable but will not be back-dated. Only the required columns are added each time to save space. **It is the responsibility of the user to ensure that the condition and value they provide are valid SQL**, but they can access any columns within the base_events_this_run_table.

This work is being done to enable the production of a g4a style marketing dashboard.

The macro will be moved into utils, and will support other warehouses, in general I want to make it a more flexible macro and possibly use a `get_array_agg` macro but our current stirng/array type macros in an effort to make them very flexible only support single column sql names, not case when statements so they would need some further work, for now it may be best to leave some of these hard-coded just for conversions (as this is the first use case and not many others use the utils package). For now, I just wanted to get it working.

Expectation is adding this to bigquery should be relatively straightforward, redshift/postgres a little more complex and databricks the most due to the lack of an order by in their array_agg function. 

1 other change, the overall session absolute time and start/end timestamp now come from ALL events, not just page pings and views.

## Checklist
- [x] I have verified that these changes work locally
- [x] I have updated the README.md (if applicable)
- [x] I have added tests & descriptions to my models (and macros if applicable)
- [x] I have raised a [documentation](https://github.com/snowplow/documentation) PR if applicable (https://github.com/snowplow/documentation/pull/425)
- [ ] Is your change a breaking change?

### CODE REVIEW

**Comment by rlh1994** [comment]
One potential limitation of the approach taken (do it all in one CTE rather than one CTE _per_ conversion) is that genuine conversions with a null value in the conversion field will not be displayed in the `_values` field. This seems unlikely to be a huge issue but could lead to a different number of conversion value to total conversions...

**Comment by rlh1994** [comment]
Another note on this is because it runs on the events table, while we have done de-duping it doesn't allow for distincting on some type of conversion id or any other form of duplicate event checking, and this would **greatly** increase the complexity to support.

File: integration_tests/dbt_project.yml [hunk 1]
```diff
@@ -271,6 +272,10 @@ seeds:
             operating_system_name: string
             operating_system_name_version: string
             operating_system_version: string
+            cv_view_page_volume: integer
+            cv_view_page_total: float
+            cv_view_page_first_conversion: timestamp
```
[the remark this answers is not in the document]
Comment (rlh1994) [reply integration_tests/dbt_project.yml:277]: The challenge with that (and the reason I didn't do first value) is that `first_value` in most warehouses isn't a pure aggregate function, only a window function, so we'd have to `row_number` first then aggregate after that was calculated. They can always list events and extract the first ID from that if needed?

File: dbt_project.yml [hunk 2]
```diff
@@ -66,6 +66,11 @@ vars:
     # Variables - Optional Consent model only
     snowplow__consent_cmp_visible: "{{ source('atomic', 'com_snowplowanalytics_snowplow_cmp_visible_1') }}"
     snowplow__consent_preferences: "{{ source('atomic', 'com_snowplowanalytics_snowplow_consent_preferences_1') }}"
+    # Variables - Optional Conversions only
+    # snowplow__conversion_events: [{'name': 'view_page', 'condition': "event_name = 'page_view'", "list_events": true},
```
Comment (miike) [inline dbt_project.yml:70]: I think we may need to consider a property that takes into account how the conversion should be counted within a given context (once per event or once per session - I don't think we need once per user).

The most important context is probably within a session i.e., should we count a type of conversion more than once in a session, or just a single time? GA has just introduced the 'legacy' counting method of once per session: https://support.google.com/analytics/answer/13366706

File: integration_tests/dbt_project.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (miike) [reply integration_tests/dbt_project.yml:277]: I think without the id is fine for the moment (in the future it probably makes sense to have a standalone conversions table that could contain this).

File: dbt_project.yml (hunk 2 above)
Comment (rlh1994) [reply dbt_project.yml:70]: Would we want this to overwrite the `_volume` and `_total` values? Currently they could get it from just a case when >0 and accessing the first value array element, but can do that to overwrite those fields if it makes more sense

**Review by agnessnowplow — APPROVED** [review]
LGTM, subject to additional fields to be added. Tested in all databases, it seemingly runs fine now for all.

**Pull request merged**